BookService/Logic/LibraryCollection.cs

```diff
@@ -0,0 +1,318 @@
+        {
+            books.Add(book);
+        }
+
+        /// <summary>
+        /// Return collection of books
+        /// </summary>
+        /// <returns>IEnumerable collection</returns>
+        public IEnumerable<Book> GetBooks()
+        {
+            return books;
+        }
+
+        /// <summary>
+        /// Remove book by it's index
+        /// </summary>
+        /// <param name="id">Index of the selected book</param>
+        /// <returns>Deleted book</returns>
+        /// <exception cref="System.IndexOutOfRangeException">Throw if index out of range</exception>
+        public Book RemoveBook(int id)
+        {
+            Book deleted = books[id];
+            books.RemoveAt(id);
+
+            return deleted;
+        }
+
+        /// <summary>
+        /// Return collection of authors
+        /// </summary>
+        /// <returns>IEnumerable collection</returns>
+        public IEnumerable<Author> GetAuthors()
+        {
+            return authors;
+        }
+
+        /// <summary>
+        /// Remove author by it's index
+        /// </summary>
+        /// <param name="id">Index of the selected book</param>
+        /// <returns>Deleted book</returns>
+        /// <exception cref="ArgumentOutOfRangeException">Throw if index out of range</exception>
+        public Author RemoveAuthor(int id)
+        {
+            Author author = authors[id];
+            bookAuthors.RemoveAll(items => authors[items.AuthorIndex] == author);
+            authors.RemoveAt(id);
+
+            return author;
+        }
+
+        /// <summary>
+        /// Method which update author reference of the selected book
+        /// </summary>
+        /// <param name="authorId">Author's index</param>
+        /// <param name="bookId">Book's index</param>
+        /// <exception cref="IndexOutOfRangeException">Throw when ether author's index
+        /// or book's index out of range of their collection's count</exception>
+        public void UpdateAuthor(int authorId, int bookId)
+        {
+            if (bookAuthors.FindAll(item => item.AuthorIndex == authorId
+                                             && item.BookIndex == bookId).Count == 0)
+            {
+                bookAuthors.Add(new BookAuthor()
+                {
+                    BookIndex = bookId, AuthorIndex = authorId
+                });
+            }
+        }
+
+        /// <summary>
+        /// Add author to collection
+        /// </summary>
+        /// <param name="author">New author</param>
+        public void AddAuthor(Author author)
+        {
+            authors.Add(author);
+        }
+
+        /// <summary>
+        /// Search books by it's genre
+        /// </summary>
+        /// <param name="genreIndex">Index of the genre</param>
+        /// <returns>Collection of books</returns>
+        public IEnumerable<Book> SearchByGenre(int genreIndex)
+        {
+            var gettedBooks = from item in bookGenres
```

Comment:
It`s odd that here you are using sql syntax and few lines up lambda syntax. Even if we will ignore my next comment regarding this, you should use 1 style per application.  But if we will consider the next comment. linq to object use lambda and for linq to sql you have options what you want to use more.